assets/js/components/AssetNavigation.js

```diff
@@ -75,6 +75,7 @@ module.exports = class AssetNavigation {
     const previous = utils.buildElement('span', ['asset-viewer-inline__previous'], '', navigation);
     if (i > 0) {
       previous.addEventListener('click', () => {
+        this.window.history.pushState(null, '', '#' + this.assetItems[i - 1].id);
```

Comment:
Avoids an edge case where the current hash is where the link is going (and so no hashchange event), eg you started on a supplement, went to the figure figure, clicked a link in the primary figure to go back to the supplement.

Reply:
This means the navigation always changes the fragment. For this actual problem, we could only care if the fragment is the ID of the asset that you're leaving, and remove it.